### PR TITLE
Add band data values to the collection metadata

### LINKED ISSUE #224: Add details about band data values to metadata
Add raster:bands object to collection metadata summaries.

## Example for S2-L2A B01 band

What we would need to add to the yaml files (for each band).

```yaml
  raster:bands:
    - name: B01
      data_type: float32
      statistics:
        minimum: 
          - 0
        maximum:
          - 0.4
 ```

How it will look rendered as json:

```json
"summaries": {
    "raster:bands": [
      {
        "name": "B01",
        "data_type": "float32",
        "statistics": {"minimum": 0, "maximum": 0.4}
      },
```

### Related resources:
- EDC public collections repository has the metadata in text/table form for some collections (see the `DocumentationLinks` we added to the links object)
- info about most of the core SH collections can be found in our documentation e.g. [here](https://docs.sentinel-hub.com/api/latest/data/sentinel-2-l2a/#units) for S2-L2A
- Info on raster STAC extension used here: https://github.com/stac-extensions/raster
- related to this request within the openEO Platform project: https://github.com/openEOPlatform/architecture-docs/issues/178#issuecomment-1006606630

### PR DESCRIPTION
**Pull request by horvatdino**
Closes #224 

**This PR:**

- Adds band data values to the public collections metadata summaries (`raster:bands object`) for the core SH collections:

  - `Copernicus DEM`
  - `Landsat 1-5 MSS L1`
  - `Landsat 4-5 TM L1, L2`
  - `Landsat 7 ETM+ L1, L2`
  - `Landsat 8-9 L1, L2`
  - `Mapzen DEM`
  - `MODIS MCD43A4.006`
  - `Sentinel-1 GRD`
  - `Sentinel-2 L1C, L2A`
  - `Sentinel-3 OLCI L1B, SLSTR L1B`
  - `Sentinel-5P L2`

### CODE REVIEW

File: collections/sentinel-1-grd.yaml [hunk 1]
```diff
@@ -221,6 +239,12 @@ Summaries:
         - - 0.00037037037
           - 0.00037037037
         unit: °
+      data_type: float32
+      statistics:
+        minimum:
+          - 0
+        maximum:
+          - 0.5
```
Comment (dthiex) [inline collections/sentinel-1-grd.yaml:247]: Can you merge latest master to this branch? The `dataMask` band we added is missing here (when doing so please also check, I think we forgot to add the band at one spot to the list of available bands.

File: collections/landsat-4-5-tm-l1.yaml [hunk 2]
```diff
@@ -291,50 +327,68 @@ Summaries:
...
         unit: m
+      data_type: INT16
     - description: Sun Azimuth Angle
       name: SAA
       openeo:gsd:
         value:
           - 30
           - 30
         unit: m
+      data_type: INT16
```
Comment (dthiex) [inline collections/landsat-4-5-tm-l1.yaml:375]: For some of the angles we have set FLOAT32. I think we should check for the Landsat collections what SH returns (decimal numbers or not) and update this accordingly.

**Review by dthiex — CHANGES_REQUESTED** [review]
A few comments as reference for later to better understood what we did/thought when doing this:

This statistics can be both a help for the user and for the backend that implements usage of those SH collections.

**Min./Max.**
- for the user helpful to know what values to expect when he needs to scale the data to png/tiff

**Datatype:**
- for the backend together with the min./max. values useful to know as how to treat the data
- for the user within openEO not really helpful/needed

**SourceFormat** from SH documentation: format in which the raw data is stored. When for example requesting S2 as reflectance data is returned in a different format (DN = 10000 * Reflectance)

File: collections/copernicus-dem.yaml [hunk 3]
```diff
@@ -136,7 +136,19 @@ Summaries:
         - - 0.000277777777778
           - 0.000277777777778
         unit: °
+      data_type: float32
```
[the remark this answers is not in the document]
Comment (horvatdino) [reply collections/copernicus-dem.yaml]: Capitalised in all collections to be consistent.

File: collections/landsat-1-5-mss-l1.yaml [hunk 4]
```diff
@@ -239,22 +257,36 @@ Summaries:
           - 60
           - 60
         unit: m
+      data_type: float32
+      statistics:
+        minimum:
+          - 0
+        maximum:
+          - 0.4
```
[the remark this answers is not in the document]
Comment (horvatdino) [reply collections/landsat-1-5-mss-l1.yaml]: Changed to 1 for all Landsat collections according to the documentation.

File: collections/landsat-4-5-tm-l2.yaml [hunk 5]
```diff
@@ -298,92 +334,141 @@ Summaries:
...
+          - 1
     - description: Pixel distance to cloud
       name: ST_CDIST
       openeo:gsd:
         value:
           - 30
           - 30
         unit: m
+      data_type: float32
+      statistics:
```
[the remark this answers is not in the document]
Comment (horvatdino) [reply collections/landsat-4-5-tm-l2.yaml:438]: This was added by mistake, was updated now.

File: collections/sentinel-1-grd.yaml (hunk 1 above)
Comment (horvatdino) [reply collections/sentinel-1-grd.yaml:249]: Merged latest master and added `dataMask` to the list of bands.

File: collections/landsat-4-5-tm-l1.yaml (hunk 2 above)
Comment (horvatdino) [reply collections/landsat-4-5-tm-l1.yaml]: Checked the SH returns and updated accordingly.

File: collections/landsat-1-5-mss-l1.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (dthiex) [reply collections/landsat-1-5-mss-l1.yaml]: I meant to only change the infrared bands (so not the rgb bands and other's as well) as you did. I think it should be fairly easy to find which are those online.

[the remark this answers is not in the document]
Comment (horvatdino) [reply collections/landsat-1-5-mss-l1.yaml]: Updated with correct statistics.

**Comment by horvatdino** [comment]
> Please see my comment and correct this.

Applied changes with statistics and found some other mistakes with data type that should be corrected now.

**Review by dthiex — APPROVED** [review]

**Pull request merged**